src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -29,53 +25,51 @@ export const configureClient = async (
   config: DataSourcePluginConfigType,
   logger: Logger
 ): Promise<Client> => {
-  const dataSource = await getDataSource(dataSourceId, savedObjects);
-  const rootClient = getRootClient(dataSource.attributes, config, openSearchClientPoolSetup);
+  try {
+    const dataSource = await getDataSource(dataSourceId, savedObjects);
+    const rootClient = getRootClient(dataSource.attributes, config, openSearchClientPoolSetup);
 
-  return getQueryClient(rootClient, dataSource, savedObjects, cryptographyClient);
+    return await getQueryClient(rootClient, dataSource, savedObjects, cryptographyClient);
+  } catch (error: any) {
+    logger.error(
+      `Fail to get data source client for dataSourceId: [${dataSourceId}]. Detail: ${error}`
+    );
+    // Re-throw as DataSourceError
+    throw new DataSourceError('Fail to get data source client: ', 400, error);
```

Comment:
The message prefix passed in here seems different from the screen shot. Can we update either?

![image](https://user-images.githubusercontent.com/44716267/187809023-9615a236-89bd-4342-bd08-e5fa57232ce5.png)

Reply:
Also shall we decorate the actual method where throws the error instead, instead of upper stream?

Reply:
added 1 line to print log on server side